inc/msp/message.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef MESSAGE_HPP
+#define MESSAGE_HPP
+
+#include "variants.hpp"
+#include "msp_id.hpp"
+#include "byte_vector.hpp"
+
+#include <vector>
+#include <string>
+#include <memory>
+#include <iostream>
+
+
+
+
+namespace msp {
+
+
+class Message {
+public:
+    virtual ID id() const = 0;
+    
+    Message() {};
+    Message(FirmwareVariant v) : fw_variant(v) {};
+    virtual ~Message() { };
+    
+    void set_fw_variant(FirmwareVariant v) 
+    {
+        fw_variant = v;
+    };
+    
+    FirmwareVariant get_fw_variant() const
+    {
+        return fw_variant;
+    };
+    
+    virtual bool decode(ByteVector &data) 
+    {
+        return false;
+    };
+    
+    virtual ByteVector_uptr encode() const
+    {
+        return ByteVector_uptr();
+    };
+    
+    virtual std::ostream& print(std::ostream& s) const
```

Comment:
Remove this print function in favour of the streaming operators in the dedicated source/header file.